ietfdata/tools/affiliations.py

```diff
@@ -0,0 +1,152 @@
+# TODO: Add licensing
+
+import sys
+import csv
+import json 
+import copy
+
+import datetime
+from datetime import timedelta
+from typing import Optional
+
+from ietfdata.datatracker     import *
+from ietfdata.datatracker_ext import *
+from ietfdata.mailarchive2    import *
+
+# load affiliation_mappings
+# import affiliations_map #hugo
+import ietfdata.tools.affiliation_mapping_dictionary as afmap #yangjun
+# affiliations.py --- script to generate extract affiliations
+# Mappings generated:
+# 1. raw affiliation -> normalised affiliation mappings
+# 2. email_domain -> normalised affiliations mappings
+# 3. identity -> start year-month, end year-month, affiliation mapping
+
+# Affiliation Entry Class
+class AffiliationEntry:
```

Comment:
Not clear what is the purpose of this class. I thought it was a record of a single affiliation with a date range, but in that case why is `names` a list?